src/pushex.ts

```diff
@@ -7,17 +7,17 @@ const DEFAULT_SOCKET_RECONNECT_ALGORITHM = (tries: number) => {
   return [3000, 6000, 10000, 20000][tries - 1] || 30000
 }
 
-type PushexConfig = {
-  getParams: () => Promise<Record<string, any>>
-  onConnect: (instance: Pushex) => void
-  onConnectionError: (instance: Pushex) => void
-  socketReconnectAlgorithm: typeof DEFAULT_SOCKET_RECONNECT_ALGORITHM
+export type PushexConfig = {
+  getParams?: () => Promise<Record<string, any>>
+  onConnect?: (instance: Pushex) => void
+  onConnectionError?: (instance: Pushex) => void
+  socketReconnectAlgorithm?: typeof DEFAULT_SOCKET_RECONNECT_ALGORITHM
 }
 
 export class Pushex {
-  private subscriptions: Record<string, Subscription>
+  private subscriptions: Record<string, Subscription> = {}
   private socket!: Socket
-  private config: PushexConfig
+  private config: Required<PushexConfig>
```

Comment:
Internally, the config will be the `PushexConfig` type but with each property always defined. The `Required` helper type does exactly that.